source/Octopus.Tentacle.Kubernetes.Tests.Integration/KubernetesScriptServiceV1IntegrationTest.cs

```diff
@@ -59,4 +112,197 @@ Task ScriptCompleted(CancellationToken ct)
             return Task.CompletedTask;
         }
     }
+    
+    [Test]
+    public async Task ScriptPodIsTerminatedDuringScriptExecution_ScriptFails()
+    {
+        // Arrange
+        var logs = new List<ProcessOutput>();
+        var scriptCompleted = false;
+        var semaphoreSlim = new SemaphoreSlim(0, 1);
+
+        var command = new ExecuteKubernetesScriptCommandBuilder(LoggingUtils.CurrentTestHash())
+            .WithScriptBody(script => script
+                .Print("Hello World")
+                .Sleep(TimeSpan.FromSeconds(1))
+                .Print("waitingtobestopped")
+                .Sleep(TimeSpan.FromSeconds(100)))
+            .Build();
+
+        //act
+        var scriptTask = Task.Run(async () => await TentacleClient.ExecuteScript(command, StatusReceived, ScriptCompleted, new InMemoryLog(), CancellationToken.None));
```

Comment:
Would it be better to pass in the test's cancellation token here instead of using `.None`? That way if something happens to the script task (it doesn't reach the status received for whatever reason) then we won't risk blocking on the semaphoreSlim indefinitely.

On a related note, I noticed that the cancellation token source in `KubernetesAgentIntegrationTest.SetUp` isn't initialised with a timeout, I think we can set something reasonable there so tests can fail within a time frame.

Reply:
Yeah, good call! I'd forgotten we had that cancellation token 😅